File: src/api/subsonic/types/baseAlbum.ts
import * as v from 'valibot';

export const BaseAlbum = v.object({
  artist: v.string(),
  artistId: v.string(),
  coverArt: v.string(),
  duration: v.number(),
  genre: v.optional(v.string()),
  genres: v.optional(v.array(v.object({ name: v.string() }))),
  id: v.string(),
  musicBrainzId: v.optional(v.string()),
  name: v.string(),
  songCount: v.number(),
  starred: v.optional(v.string()),
  userRating: v.optional(v.number()),
  year: v.optional(v.number()),
});

export type BaseAlbum = v.InferInput<typeof BaseAlbum>;

File: src/api/subsonic/baseAlbum.ml
open Ppx_deriving_json_runtime.Primitives

type[@deriving json] t = {
  artist : string;
  artistId : string;
  coverArt : string;
  duration : float option; [@json.option]
  genre : string option; [@json.option]
  genres : Genre.t array option; [@json.option]
  id : string;
  musicBrainzId : string option; [@json.option]
  name : string;
  songCount : int;
  starred : string option; [@json.option]
  userRating : int option; [@json.option]
  year : int option; [@json.option]
}
